### PR TITLE
OHRM5X-1553 Broken issues

### PR DESCRIPTION
**Pull request by SharukaNP**

### CODE REVIEW

File: src/client/src/orangehrmPerformancePlugin/components/ReviewConfirmModal.vue [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <teleport to="#app">
+    <oxd-dialog
+      v-if="show"
+      class="orangehrm-dialog-popup"
+      :style="{maxWidth: '450px'}"
```
Comment (RajithaKumara) [inline src/client/src/orangehrmPerformancePlugin/components/ReviewConfirmModal.vue:26]:
```suggestion
```

File: installer/Migration/V5_1_0/lang-string/performance.yaml [hunk 2]
```diff
@@ -233,3 +233,9 @@ langStrings:
   -
     value: 'General Comment'
     unitId: general_comment
+  -
+    value: 'Confirm Review Submission'
+    unitId: confirm_review_submission
+  -
+    value: 'The Review will be made read-only after completion. This action cannot be undone. Are you sure you want to undone'
```
Comment (RajithaKumara) [inline installer/Migration/V5_1_0/lang-string/performance.yaml:240]:
```suggestion
    value: 'The review will be made read-only after completion. This action cannot be undone. Are you sure you want to complete?'
```
:thinking:

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 3]
```diff
@@ -385,37 +385,44 @@ public function getReviewById(int $id): ?PerformanceReview
...
         return $qb->getQuery()->execute();
     }
 
     /**
      * @param ReviewKpiSearchFilterParams $reviewKpiSearchFilterParams
+     * @param string $reviewerGroupName
      * @return int
      */
-    public function getKpisForReviewCount(ReviewKpiSearchFilterParams $reviewKpiSearchFilterParams): int
+    public function getKpisForReviewCount(ReviewKpiSearchFilterParams $reviewKpiSearchFilterParams, string $reviewerGroupName): int
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:402]:
```suggestion
    public function getKpisForReviewCount(ReviewKpiSearchFilterParams $reviewKpiSearchFilterParams): int
```
better to move `$reviewerGroupName` into `ReviewKpiSearchFilterParams`

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 4]
```diff
@@ -385,37 +385,44 @@ public function getReviewById(int $id): ?PerformanceReview
...
+    private function getKpisForReviewQueryBuilderWrapper(ReviewKpiSearchFilterParams $reviewKpiSearchFilterParams, string $reviewerGroupName): QueryBuilderWrapper
     {
-        $jobTitleId = $this->getReviewById($reviewKpiSearchFilterParams->getReviewId())->getJobTitle()->getId();
-        $q = $this->createQueryBuilder(Kpi::class, 'kpi');
-        $q->andWhere('kpi.jobTitle =:jobTitle')
-            ->setParameter('jobTitle', $jobTitleId);
-        $this->setSortingAndPaginationParams($q, $reviewKpiSearchFilterParams);
-
-        return $this->getQueryBuilderWrapper($q);
+        $reviewerGroup = $this->getRepository(ReviewerGroup::class)->findOneBy(['name' => $reviewerGroupName]);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:415]: will left join in query builder
```suggestion
```

File: src/plugins/orangehrmPerformancePlugin/Api/Model/KpiSummaryModel.php [hunk 5]
```diff
@@ -21,21 +21,21 @@
 
 use OrangeHRM\Core\Api\V2\Serializer\ModelTrait;
 use OrangeHRM\Core\Api\V2\Serializer\Normalizable;
-use OrangeHRM\Entity\Kpi;
+use OrangeHRM\Entity\ReviewerRating;
 
 class KpiSummaryModel implements Normalizable
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/Model/KpiSummaryModel.php:26]: did you check the usage of `KpiSummaryModel` class?

File: src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue [hunk 6]
```diff
@@ -207,12 +208,14 @@ export default {
...
-            reviewer?.terminationId ? ` ${$t('general.past_employee')}` : ''
-          }`,
+          employee: $tEmpName(employee, {
+            includeMiddle: false,
+            excludePastEmpTag: false,
+          }),
+          reviewer: $tEmpName(reviewer, {
+            includeMiddle: false,
+            excludePastEmpTag: false,
+          }),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue:218]:
```suggestion
          reviewer: $tEmpName(reviewer),
```
:thinking: check like this

File: src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue [hunk 7]
```diff
@@ -207,12 +208,14 @@ export default {
...
-          employee: `${employee?.firstName} ${employee?.lastName} ${
-            employee?.terminationId ? ` ${$t('general.past_employee')}` : ''
-          }`,
-          reviewer: `${reviewer?.firstName} ${reviewer?.lastName} ${
-            reviewer?.terminationId ? ` ${$t('general.past_employee')}` : ''
-          }`,
+          employee: $tEmpName(employee, {
+            includeMiddle: false,
+            excludePastEmpTag: false,
+          }),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue:214]:
```suggestion
          employee: $tEmpName(employee),
```
@devishke-orange both options are false by default right?

File: src/client/src/orangehrmPerformancePlugin/components/EvaluationForm.vue [hunk 8]
```diff
@@ -229,7 +229,7 @@ export default {
     const employeeName = computed(() => {
       return $tEmpName(props.employee, {
         includeMiddle: false,
-        excludePastEmpTag: true,
+        excludePastEmpTag: false,
       });
```
Comment (RajithaKumara) [inline src/client/src/orangehrmPerformancePlugin/components/EvaluationForm.vue:233]:
```vue 
return $tEmpName(props.employee);
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue (hunk 7 above)
Comment (devishke-orange) [reply src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue:214]: yes no need to have the options object in this case

**Review by Super-Chama — APPROVED** [review]

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php (hunk 3 above)
Comment (SharukaNP) [reply src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:402]: Fixed but will push once overall testing on local runs and finish

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**